src/Symfony/Component/Utf8/CodePoints.php

```diff
@@ -0,0 +1,390 @@
+        $result->string = preg_replace("/[$charsList]+\$/u", '', $this->string);
+
+        return $result;
+    }
+
+    public function getIterator(int $maxChunkLength = 1): \Generator
+    {
+        if ($maxChunkLength < 1) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
+        }
+
+        if ('' === $this->string) {
+            return;
+        }
+
+        if ($maxChunkLength > 65535) {
+            $maxChunkLength = 65535;
+        }
+
+        foreach (preg_split('/(.{'.$maxChunkLength.'})/u', $this->string, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY) as $key => $char) {
+            $clone = clone $this;
+            $clone->string = $char;
+            yield $key => $clone;
+        }
+    }
+
+    public function append(string $suffix): self
+    {
+        $result = clone $this;
+        $result->string.= self::fromString($suffix)->string;
+
+        return $result;
+    }
+
+    public function prepend(string $prefix): self
+    {
+        $result = self::fromString($prefix);
+        $result->string.= $this->string;
+
+        return $result;
+    }
+
+    public function toBytes(): Bytes
+    {
+        return Bytes::fromString($this->string);
+    }
+
+    public function toGraphemes(): Graphemes
+    {
+        return Graphemes::fromString($this->string);
+    }
+
+    public function indexOf(string $needle, int $offset = 0)
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        if (PHP_VERSION_ID < 70100 && $offset < 0) {
+            return null;
+        }
+
+        $result = mb_strpos($this->string, $needle, $offset > 0 ? $offset : 0);
+
+        return false === $result ? null : $result;
+    }
+
+    public function indexOfIgnoreCase(string $needle, int $offset = 0)
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        if (PHP_VERSION_ID < 70100 && $offset < 0) {
+            return null;
+        }
+
+        $result = mb_stripos($this->string, $needle, $offset);
+
+        return false === $result ? null : $result;
+    }
+
+    public function lastIndexOf(string $needle, int $offset = 0)
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        $result = mb_strrpos($this->string, $needle, $offset);
+
+        return false === $result ? null : $result;
+    }
+
+    public function lastIndexOfIgnoreCase(string $needle, int $offset = 0)
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        $result = mb_strripos($this->string, $needle, $offset);
+
+        return false === $result ? null : $result;
+    }
+
+    public function substringOf(string $needle, int $offset = 0)
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        $haystack = $offset > 0 ? mb_substr($this->string, $offset, null, 'UTF-8') : $this->string;
+
+        $result = mb_strstr($haystack, $needle, null, 'UTF-8');
```

Comment:
should be $beforeNeedle, new arg on the mthod in fact, isn't it? (same for methods below)